### DISCUSSION TITLE
Factory reset?

### DISCUSSION THREAD

**Discussion by angusmci** [Phambili-Tech/Newt_Display#29] [Q&A] [answered]
How do you factory reset a Newt?

My Newt initially started up successfully. However, after I connected to the WiFi config screen, selected my WiFi network and entered the password, it began looping on startup. It displays "Welcome to Newt", with 'Connecting to WiFi', followed by 'Initializing Device' and 'Version 0.0.11', then shows the Attribution screen, waits for a moment, then returns to the Welcome screen.

Pressing or pressing and holding the boot and reset buttons, together or separately, doesn't seem to resolve the issue.

The local access point doesn't show up in the list of available networks any more, so I can't connect and try re-entering the network password.

**Comment by darianbjohnson** [comment]
Hey, I’m out of town (and away from my PC) for the weekend; I’ll ping you when I’m back to troubleshoot the problem

**Comment by jhoskins98** [comment]
Curious - have you tried using your phone as a hot spot for WiFi? What type of router are you using? Is it a mesh setup?

**Reply by jbatesmd** [reply]
I used my phone and I am having the same issue.  Continually resets never allowing an update.  Is there a newer version of the firmware my is 0.0.11

**Reply by jondaddio** [reply]
Same here, constant looping never allowing an update.  Mine was flawless until this morning.  Hung up on one time (10:21), and when reset went into the endless loop.  Occasionally provides the time, 6 hours off.

**Comment by darianbjohnson** [comment, marked as answer]
There is a newer version. You should be able to download/install manually. 

1. Download the latest version of the code (v 1.0.2) here -> [https://phambili-pub.s3.amazonaws.com/Newt.ino_latest.bin](https://phambili-pub.s3.amazonaws.com/Newt.ino_latest.bin)

2. Plug your newt into a PC/computer and hit the reset button. A USB Drive should appear (a file named "Firmware.bin" will be in the drive).

3. Drag the file you downloaded (Newt.ino_lastest.bin) onto the USB drive (same location as the Firmware.bin file).

The Newt should restart with version 1.0.2. If you have wifi problems still, then go to the settings menu and and select "Reset Wifi"

Let me know if you are having problems. Newt should be caught the restart loop that you're experiencing...

**Reply by angusmci** [reply]
That seems to have taken care of it. I had some minor difficulty because the constant restart cycle kept unmounting the drive, but eventually I got the new firmware copied across before it restarted, and now it seems to work as it should. Thank you.

**Reply by jondaddio** [reply]
My Newt is in a constant loop. How did you get the USB drive to appear?

**Reply by darianbjohnson** [reply]
@jondaddio

- Plug your Newt into a PC/Computer, then push the reset button (left button if Newt screen is facing you; right button if Newt screen if facing away from you). 
- When you press the button, then a USB drive should appear

If it appears, but then disappears before uploading the code (because of the reset loop), then I can send you a small "dummy" file that will stop the reset

If it does appear - let me know. If it's not showing up, then maybe there is some "weirdness" on the back end (Newt does an initial wifi call when it starts up for the first time to get basic information- maybe something is causing a problem...).

**Reply by jondaddio** [reply]
Well, I got out my USB-A to USB-C cable tester,  Turned out the first cable I used was only good for power -- which is why the USB Drive never showed up on Explorer.  Once I found my what-I-thought-was-a good cable, it tested good for power and USB 2.x.  I plugged that into the Newt, plugged that into my PC, and then the USB Drive showed up.  I dragged the Newt.ino-latest.bin file onto the USB Drive, and it quickly updated the Newt.  And the Newt powered right up and worked normally.

Now all is well.  So the manual update fixed the endless looping for me.  (Just remember to test your USB-A to USB-C cable to make sure it passes data !!!)

**Reply by tjfoth** [reply]
@darianbjohnson I have the reset loop.  Would you please send me the dummy file? 

Thanks!

Tom

**Reply by darianbjohnson** [reply]
@tjfoth  Sorry that your device is stuck in a reset loop (I think I need to write a piece of code that skips the wifi call if we see repeated reboots). Anyway, I turned the "Hello World" file into a bin. Download it here - https://phambili-pub.s3.amazonaws.com/Example1-Hello_World.ino.esp32s2usb.bin

This will "pause" the reset loop and should allow you to reinstall the v1.1.0 or v1.0.2 code. Let me know...

**Comment by jbatesmd** [comment]
This fixed it for me as well.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by tjfoth** [comment]
I thought that might be the case, but I wasn't sure if you had some sort of
retry timer in your code that you were resetting with code...

---
Tom Foth
***@***.***
***@***.***

**Reply by tjfoth** [reply]
I'm good! It is up and running!

**Comment by gshafer** [comment]
I am stuck in the loop. I tried to install the "Hello World" app and that stopped it. When I tried to install the latest .bin file, it just went right back into the loop. Very frustrating. I have 2 of these units --- both stuck in the loop and I cannot use them for anything!

**Reply by darianbjohnson** [reply]
@gshafer - I'm sorry that your having this problem; I've never seen a unit (much less two owned by the same user) to stay stuck in the reset loop. Ok - can you use this  form -> https://www.crowdsupply.com/phambili/newt/ask-question - to send me your email address? I'd like to work with you to get to the bottom of this problem. In the meantime, I am going to create/compile a one-off version of the code with some "print to screen" statements that will help us to diagnose the problem.

Three additional questions:
1) has this been the case since you first started using the unit, or just since your attempted an upgrade (or did it just start randomly)?
2) Are you using a LIPO battery or USB to power the device?
3) Are you using a mesh router? or a regular router?

I think this "loop" starts when something unstable happens with the wifi connection (either a power surge that causes the ESP32 to reset, or maybe I don't have the right error logic in the connection loop). The problem is intermittent, so I'm having a hell of a time trying to diagnose the problem. 

My other hypothesis is that I'm not appropriately handing the download/initialization details (e.g. - maybe there is a null value in the location, or maybe the location string is longer than the variable char string that I'm using...). Again, the problem seems to come and go (with the same user/device/location) so that doesn't feel like the issue...

**Reply by darianbjohnson** [reply]
hey @gshafer - I'm tracking this as issue #48 ... I have a few ideas about why the issue is occurring..... continuing to troubleshoot
